kernelci/data/kernelci_api.py

```diff
@@ -83,6 +92,49 @@ def get_nodes_by_commit_hash(self, commit_hash):
     def get_node_from_event(self, event):
         return self.get_node(event.data['id'])
 
+    def pubsub_event_filter(self, sub_id, event):
+        """Filter Pub/Sub events
+
+        Filter received Pub/Sub event using provided filter dictionary.
+        Return True if the event matches with the filter, otherwise False.
+        """
+        event_filter_status = True
+        filters = self._filters.get(sub_id)
+        if not filters:
+            return event_filter_status
+        for key, value in filters.items():
+            if key not in event.keys():
+                continue
+            if isinstance(value, dict):
+                for sub_key, sub_value in value.items():
+                    if sub_key not in event.get(key):
+                        continue
+                    if sub_value != event.get(key).get(sub_key):
+                        return
+            elif value != event[key]:
+                return
+
+        return event
+
+    def receive_node(self, sub_id):
+        """
+        Listen to all the events on 'node' channel and apply filter on it.
+        Return node if event matches with the filter.
+        """
+        path = '/'.join(['listen', str(sub_id)])
+        try:
+            while True:
+                resp = self._get(path)
+                event = from_json(resp.json().get('data'))
+                node = self.get_node_from_event(event)
+                node['op'] = event.data['op']
+                if self.pubsub_event_filter(sub_id, node):
+                    return node
+        except Exception as e:
+            print(f"Error: {e}")
+        except KeyboardInterrupt as e:
+            print(f"Stopping.")
```

Comment:
Please remove this or we won't be able to abort the applications (e.g. trigger.py etc.) which relies on catching it. In fact I think we can drop all exception handling here, they can be caught by the code calling this method.

Reply:
Yes, I missed that scenario. Thanks for pointing it out.